### PR TITLE
feat(components/modals): add inline help support for modal component

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2276326](https://blackbaud.visualstudio.com/Products/_boards/board/t/SKY%20UX%20Components/Stories/?workitem=2276326)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/530/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/530/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/530/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/530/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/530/storybooks/pages-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/530/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/530/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/530/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/530/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/530/playground/)

File: libs/components/modals/src/lib/modules/modal/modal-header.component.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
   [skyThemeClass]="{
     'sky-font-display-3': 'modern'
   }"
+  skyTrim
 >
   <ng-content></ng-content>
 </h1>
+<ng-content select=".sky-control-help"></ng-content>
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/modals/src/lib/modules/modal/modal-header.component.html]: You need to collapse the whitespace around the `ng-content` as well as wrap it in a `span` with a `sky-control-help-container` class. See [description list](https://github.com/blackbaud/skyux/blob/main/libs/components/layout/src/lib/modules/description-list/description-list-term.component.html) for an example.

File: libs/components/modals/src/lib/modules/modal/modal-header.component.scss [hunk 2]
```diff
@@ -5,6 +5,13 @@
 h1 {
   margin: 0;
   line-height: 1.2;
+  display: inline;
+}
+
+::ng-deep {
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/modals/src/lib/modules/modal/modal-header.component.scss]: You won't need this once you use the `sky-control-help-container` class to wrap the `.sky-control-help` content.

File: libs/components/modals/src/lib/modules/modal/modal.interface.ts [hunk 3]
```diff
@@ -62,6 +62,7 @@ export interface SkyModalConfigurationInterface {
   tiledBody?: boolean;
 
   /**
+   * @deprecated Use SkyHelpInlineComponent in the modal header instead.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/modals/src/lib/modules/modal/modal.interface.ts:65]: I don't think we want to call out `SkyHelpInlineComponent` specifically. (It's a viable option, but it's not the only option a consumer can take to add inline help. They can use any element so long as it has the `.sky-control-help` class on it.) Take a look at the `SkyInputBoxComponent` JSDoc comment to see how we can word this to be more generic.

File: apps/code-examples/src/app/code-examples/modals/inline-help/modal-demo.component.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  #modalSvc: SkyModalService;
+
+  constructor(modalSvc: SkyModalService) {
+    this.#modalSvc = modalSvc;
+  }
+
+  public onOpenModalClick(): void {
+    const modalInstanceType: any = ModalDemoModalComponent;
+    const options: any = {
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/modals/inline-help/modal-demo.component.ts:21]: ☝️ Possible to avoid the `any` type?

**Pull request closed without merging**